Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm

```diff
@@ -338,6 +340,58 @@ static uint32_t convertSystemLayoutDirection(NSUserInterfaceLayoutDirection dire
 
 #endif // PLATFORM(MAC)
 
+#if ENABLE(SCREEN_TIME)
+- (void)_installScreenTimeWebpageController
+{
+    if (!_page->preferences().screenTimeEnabled())
+        return;
+    if (!_screenTimeWebpageController) {
+        _screenTimeWebpageController = adoptNS([[STWebpageController alloc] init]);
+        [_screenTimeWebpageController addObserver:self forKeyPath:@"URLIsBlocked" options:(NSKeyValueObservingOptionOld | NSKeyValueObservingOptionNew) context:&screenTimeWebpageControllerBlockedKVOContext];
+
+#if PLATFORM(MAC)
+        NSView *screenTimeView = [_screenTimeWebpageController view];
+        screenTimeView.frame = self.bounds;
+        screenTimeView.translatesAutoresizingMaskIntoConstraints = NO;
+        // add to webview
+        [self addSubview:screenTimeView];
+        // adjust based on constraints
+        [NSLayoutConstraint activateConstraints:@[
+            [screenTimeView.widthAnchor constraintEqualToAnchor:self.widthAnchor],
+            [screenTimeView.heightAnchor constraintEqualToAnchor:self.heightAnchor],
+            [screenTimeView.leadingAnchor constraintEqualToAnchor:self.leadingAnchor],
+            [screenTimeView.topAnchor constraintEqualToAnchor:self.topAnchor]
+        ]];
+#endif // PLATFORM(MAC)
+    }
+}
+#endif // ENABLE(SCREEN_TIME)
```

Comment:
why end the block and then reopen with the same condition?

Reply:
Will change!